lib/action_policy/testing.rb

```diff
@@ -13,14 +13,15 @@ def initialize(policy, rule)
           @rule = rule
         end
 
-        def matches?(policy_class, actual_rule, target)
+        def matches?(policy_class, actual_rule, target, context)
           policy_class == policy.class &&
             (target.is_a?(Class) ? target == policy.record : target === policy.record) &&
-            rule == actual_rule
+            rule == actual_rule && policy.authorization_context >= context
```

Comment:
What about extracting context matching into a separate method?

That would allow us to handle `nil` as well as support RSpec composed matchers (`a_hash_including(some_key: an_instance_of(SomeClass))`). For that, we can use the following check:

```ruby
context === policy.authorization_context || policy.authorization_context >= context
```

Reply:
Sure, we can extract the context matching to a particular method. But I don't understand how this code will handle when `context` is nil because it'll raise the following error
```ruby
TypeError:
        Failure/Error:
        policy_class == policy.class &&
          (target.is_a?(Class) ? target == policy.record : target === policy.record) &&
          rule == actual_rule && (context === policy.authorization_context || policy.authorization_context >= context)

      TypeError:
        no implicit conversion of nil into Hash
```

From what I understood, the `policy.authorization_context` would always be a hash, and if the context is nil as default, it'll cause the error above to happen.
https://github.com/palkan/action_policy/blob/e709556d88a434ef3b3bb1acc6afb65e48df97f0/lib/action_policy/behaviour.rb#L60-L69

We can add a guard-clause to our extracted method to prevent that error, but it can be more complex than just defaulting to an empty hash. Maybe I'm not seeing your point.

Reply:
> We can add a guard-clause

Yeah, I didn't include this into the snippet; the full version would be:

```ruby
def context_matches?(context, actual)
  return true unless context

  context === actual || actual >= context
end
```